src/native/corehost/hostmisc/utils.cpp

```diff
@@ -350,11 +350,20 @@ bool try_stou(const pal::string_t& str, unsigned* num)
 
 pal::string_t get_dotnet_root_env_var_name()
 {
+    pal::string_t dotnet_root_arch = _X("DOTNET_ROOT_"), path;
+    dotnet_root_arch.append(to_upper(get_arch()));
+    if (pal::getenv(dotnet_root_arch.c_str(), &path))
```

Comment:
Since this has exactly one callsite - we should change this to `get_dotnet_root_from_env_var` and call `get_file_path_from_env` instead of `getenv` to avoid duplicate calls to `getenv`.

This is a perf critical path for framework dependent apps (every app will run through this).